### PR TITLE
Move client to stable

### PR DESCRIPTION
**Pull request by oskarth**
This PR moves client spec from draft to stable. This means it is an accurate description of what is currently in production for the v1 app.

If we want to do wider changes in the future these should come in the form of SIP issues and PRs, as indicated in the README.

Changes to this spec can still occur but they should be cosmetic ones, errata and clarifications.

Please review with anything factually incorrect or add any relevant data. I want to get this merged over the next 24-48h. This will ensure we have a stable set of specs that accurately reflects what is in production, and then we can proceed from there.

### CODE REVIEW

**Review by decanus — APPROVED** [review]

**Review by adambabik — APPROVED** [review]

**Review by corpetty — CHANGES_REQUESTED** [review]
questions/suggestions to discuss before my approval: 
- ~~L80 - we DO NOT run on the public Ethereum network, we just use their protocols to make our own network~~
- ~~L86 - remove `Ethereum`~~
- ~~L93 - remove `Ethereum`~~
- ~~L104-106 - Don't we need both node types 2 and 4, not just mobile clients? More clarity here~~
- ~~L110 - we use the work `need` here.  Is this true? I don't think it is, but if it is, then we need to amend the previous `MUST` on L104-106, and update this to `MUST`~~
- ~~L133 - `need` -> `MUST`~~
- ~~L159 - `recommended -> `RECOMMEND` and appropriate grammar~~
- ~~L166 - use `MAY` somewhere~~
- ~~L176-177 - remove and add link to fleet list above where bootstraps are listed~~
- ~~L222 - link to Appendix A~~
- ~~L248 - any other updated libs/packages to link to here?~~ (no there isn't)

File: status-client-spec.md [hunk 1]
```diff
@@ -111,18 +114,18 @@ nodes allow you to discover other nodes of the network.
...
-  - `enode://436cc6f674928fdc9a9f7990f2944002b685d1c37f025c1be425185b5b1f0900feaf1ccc2a6130268f9901be4a7d252f37302c8335a2c1a62736e9232691cc3a@178.128.138.128:443`
-  - `enode://5395aab7833f1ecb671b59bf0521cf20224fe8162fc3d2675de4ee4d5636a75ec32d13268fc184df8d1ddfa803943906882da62a4df42d4fccf6d17808156a87@178.128.140.188:443`
-- North America:
-  - `enode://32ff6d88760b0947a3dee54ceff4d8d7f0b4c023c6dad34568615fcae89e26cc2753f28f12485a4116c977be937a72665116596265aa0736b53d46b27446296a@34.70.75.208:443`
-  - `enode://5405c509df683c962e7c9470b251bb679dd6978f82d5b469f1f6c64d11d50fbd5dd9f7801c6ad51f3b20a5f6c7ffe248cc9ab223f8bcbaeaf14bb1c0ef295fd0@35.223.215.156:443`
-
-These bootstrap nodes do not change, however, we can't guarantee that it will stay this way forever
+Status maintains a list of production fleet boootstrap nodes in the following locations:
+- Hong Kong: 
+   - "enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443"
```
Comment (jakubgs) [inline status-client-spec.md]: Maybe use backticks instead of double quotes for those?

Comment (oskarth) [inline status-client-spec.md]:
```suggestion
   - `enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443`
```

Comment (oskarth) [reply status-client-spec.md]: done

**Comment by oskarth** [comment]
@corpetty Thanks for the feedback! Could you please do the review in-line? The line numbers are quite confusing and they also change. It is much easier to have a conversation and do suggestions etc in a specific locale. E.g. I noticed you pushed a commit with fixes, but I have no idea what that fixes of your comments.

I know the file diff is a bit off, so you gotta expand to see lines not changed. Even that doesn't work all the time. I suggest pushing a commit with the comment in-line at the appropriate place, then we can comment and discuss on it in-line. WDYT?

**Comment by jakubgs** [comment]
```md
3. Mailservers (servers and clients)
```
What does `servers and clients` mean? A "mailserver" is just a server.
```
we RECOMMEND you implement a mailserver client mode
```
So isn't a `mailserver client mode` just a mobile node? I don't get this terminology.

```
`eth.beta` is the current group of peers
```
No, it's `eth.prod`.

There's still two `TODO` comments in the text. And why are you breaking lines yourself? Looks weird.

**Comment by corpetty** [comment]
> And why are you breaking lines yourself?

pretty sure this is a "tabs vs spaces" type argument

File: status-client-spec.md [hunk 2]
```diff
@@ -116,14 +116,14 @@ run these provided they are connected to the rest of the Whisper network.
 
 Status maintains a list of production fleet boootstrap nodes in the following locations:
 - Hong Kong:
```
Comment (decanus) [inline status-client-spec.md]:
```suggestion
**Hong Kong:**
```

File: status-client-spec.md [hunk 3]
```diff
@@ -116,14 +116,14 @@ run these provided they are connected to the rest of the Whisper network.
 
 Status maintains a list of production fleet boootstrap nodes in the following locations:
 - Hong Kong: 
-   - "enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443"
-   - "enode://23d0740b11919358625d79d4cac7d50a34d79e9c69e16831c5c70573757a1f5d7d884510bc595d7ee4da3c1508adf87bbc9e9260d804ef03f8c1e37f2fb2fc69@47.52.106.107:443"
+   - `enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443`
+   - `enode://23d0740b11919358625d79d4cac7d50a34d79e9c69e16831c5c70573757a1f5d7d884510bc595d7ee4da3c1508adf87bbc9e9260d804ef03f8c1e37f2fb2fc69@47.52.106.107:443`
 - Amsterdam:
```
Comment (decanus) [inline status-client-spec.md]:
```suggestion
**Amsterdam:**
```

File: status-client-spec.md [hunk 4]
```diff
@@ -116,14 +116,14 @@ run these provided they are connected to the rest of the Whisper network.
...
-   - "enode://23d0740b11919358625d79d4cac7d50a34d79e9c69e16831c5c70573757a1f5d7d884510bc595d7ee4da3c1508adf87bbc9e9260d804ef03f8c1e37f2fb2fc69@47.52.106.107:443"
+   - `enode://6e6554fb3034b211398fcd0f0082cbb6bd13619e1a7e76ba66e1809aaa0c5f1ac53c9ae79cf2fd4a7bacb10d12010899b370c75fed19b991d9c0cdd02891abad@47.75.99.169:443`
+   - `enode://23d0740b11919358625d79d4cac7d50a34d79e9c69e16831c5c70573757a1f5d7d884510bc595d7ee4da3c1508adf87bbc9e9260d804ef03f8c1e37f2fb2fc69@47.52.106.107:443`
 - Amsterdam: 
-   - "enode://436cc6f674928fdc9a9f7990f2944002b685d1c37f025c1be425185b5b1f0900feaf1ccc2a6130268f9901be4a7d252f37302c8335a2c1a62736e9232691cc3a@178.128.138.128:443"
-    - "enode://5395aab7833f1ecb671b59bf0521cf20224fe8162fc3d2675de4ee4d5636a75ec32d13268fc184df8d1ddfa803943906882da62a4df42d4fccf6d17808156a87@178.128.140.188:443"
+   - `enode://436cc6f674928fdc9a9f7990f2944002b685d1c37f025c1be425185b5b1f0900feaf1ccc2a6130268f9901be4a7d252f37302c8335a2c1a62736e9232691cc3a@178.128.138.128:443`
+    - `enode://5395aab7833f1ecb671b59bf0521cf20224fe8162fc3d2675de4ee4d5636a75ec32d13268fc184df8d1ddfa803943906882da62a4df42d4fccf6d17808156a87@178.128.140.188:443`
 - Central US:
```
Comment (decanus) [inline status-client-spec.md]:
```suggestion
**Central US:**
```

**Comment by oskarth** [comment]
Answered mailserver/client in Discord. Added eth.prod and removed TODOs

File: status-client-spec.md [hunk 5]
```diff
@@ -78,20 +78,20 @@ We use [`protobuf`](https://developers.google.com/protocol-buffers/) in differen
 
 ### P2P Overlay
 
-Status clients run on the public Ethereum network, as specified by the devP2P
+Status clients run on a private peer-to-peer network, as specified by the devP2P
```
Comment (oskarth) [inline status-client-spec.md]: Is this accurate? It is permissionless, nothing is stopping anyone from running a node. @corpetty

File: status-client-spec.md [hunk 6]
```diff
@@ -130,7 +130,7 @@ and at some point we might be forced to change them.
 
 #### Discovery
 
-To implement a Status client you need to discover peers to connect to. We use a
+To implement a Status client you MUST discover peers to connect to. We use a
```
Comment (oskarth) [inline status-client-spec.md]: Is this accurate? If you have nodes ready as static nodes you don't necessarily need to discover new nodes. SHOULD?

cc @adambabik

**Comment by corpetty** [comment]
How do we suggest cleaning up the `Node discovery and roles` section as that seems to be the most unclear?

File: status-client-spec.md [hunk 7]
```diff
@@ -156,14 +156,14 @@ Status nodes that want to be discovered MUST register to Discovery V5 and/or Ren
 with the `whisper` topic. Status nodes that are mail servers and want to
 be discoverable MUST additionally register with the `whispermail` topic.
 
-The recommended strategy is to use both mechanisms but at the same time implement a structure
+It is RECOMMENDED to use both mechanisms but at the same time implement a structure
```
Comment (oskarth) [inline status-client-spec.md:165]: Is this an official term as per RFC?

File: status-client-spec.md (hunk 5 above)
Comment (corpetty) [reply status-client-spec.md]: good point, maybe `separate, permissionless` instead of `private`?

File: status-client-spec.md [hunk 8]
```diff
@@ -216,9 +222,7 @@ see [Status EIPs Standards](status-EIPs.md).
 
 ## Security Considerations
 
-TBD.
-
-<!-- TODO: Fill this out. -->
+See [Appendix A](#appendix-a-security-considerations)
```
Comment (oskarth) [inline status-client-spec.md:228]: This link seems weird?

File: status-client-spec.md (hunk 5 above)
Comment (corpetty) [reply status-client-spec.md]: or just `permissionless`

File: status-client-spec.md (hunk 6 above)
Comment (corpetty) [reply status-client-spec.md]: have a list of static nodes is trivial discovery, imo

File: status-client-spec.md (hunk 5 above)
Comment (oskarth) [reply status-client-spec.md]: I'd say public and permissionless

File: status-client-spec.md [hunk 9]
```diff
@@ -75,20 +78,20 @@ We use [`protobuf`](https://developers.google.com/protocol-buffers/) in differen
 
 ### P2P Overlay
 
-Status clients run on the public Ethereum network, as specified by the devP2P
+Status clients run on a private peer-to-peer network, as specified by the devP2P
```
Comment (oskarth) [inline status-client-spec.md]:
```suggestion
Status clients run on a public, permissionless peer-to-peer network, as specified by the devP2P
```

File: status-client-spec.md (hunk 7 above)
Comment (corpetty) [reply status-client-spec.md:165]: > 3. SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.

File: status-client-spec.md [hunk 10]
```diff
@@ -111,23 +114,23 @@ nodes allow you to discover other nodes of the network.
...
+   - `enode://32ff6d88760b0947a3dee54ceff4d8d7f0b4c023c6dad34568615fcae89e26cc2753f28f12485a4116c977be937a72665116596265aa0736b53d46b27446296a@34.70.75.208:443`
+   - `enode://5405c509df683c962e7c9470b251bb679dd6978f82d5b469f1f6c64d11d50fbd5dd9f7801c6ad51f3b20a5f6c7ffe248cc9ab223f8bcbaeaf14bb1c0ef295fd0@35.223.215.156:443`
+
+These bootstrap nodes MAY change and we can't guarantee that it will stay this way forever
 and at some point we might be forced to change them.
 
 #### Discovery
 
-To implement a Status client you need to discover peers to connect to. We use a
+To implement a Status client you MUST discover peers to connect to. We use a
```
Comment (oskarth) [inline status-client-spec.md]:
```suggestion
To implement a Status client you MUST discover or have a list of peers to connect to. We use a
```

File: status-client-spec.md (hunk 8 above)
Comment (corpetty) [reply status-client-spec.md:228]: why? it just goes to the appendix at the bottom of the page

Comment (oskarth) [reply status-client-spec.md:228]: ah, my bad, this is middle

**Comment by corpetty** [comment]
This doc lacks most uses outside of chat, should they be included?

**Comment by oskarth** [comment]
> This doc lacks most uses outside of chat, should they be included?

We probably should, at least mention them and then full specs can come later. E.g. we need the wallet tx stuff way more specified

**Comment by corpetty** [comment]
so we have:
- account management
- chat
- wallet
- web3 browser

Correct?

**Comment by corpetty** [comment]
@oskarth I made what I think to be the last edit for this.  Please review that commit and merge if you feel it is sufficient.

File: README.md [hunk 11]
```diff
@@ -15,17 +15,21 @@ Every spec has its own lifecycle that shows its maturity. We indicate this in a
...
 The following SIPs are under consideration for standardization.
 
-- [Status Client Specification](status-client-spec.md). The main specification for writing a Status client. **Start here**
 - [Status Secure Transport Specification](status-secure-transport-spec.md). How Status provide a secure transport with conversational security properties.
 - [Status Payload Specification](status-payloads-spec.md). What the message payloads look like.
 - [Status Account Specification](status-account-spec.md). What a Status account is and how trust is established.
 - [Status Whisper Usage Specification](status-whisper-usage-spec.md). How we use Whisper to do routing, metadata protection and provide 1:1/group/public chat.
 - [Status Whisper Mailserver Specification](status-whisper-mailserver-spec.md). How we use Whisper mailservers to provide offline inboxing.
 - [Status EIPs Standards](status-EIPs.md). Ethereum Improvement Proposals used in Status.
+- [Status Blockchain Usage Specification](./status-blockchain-spec.md). How we use the Ethereum blockchain to facilitate the Web3 Browser and Wallet functionality
```
Comment (oskarth) [inline README.md:42]: These aren't draft, they are barely raw at this stage. I'd remove these and then we can include as draft once there is something here

File: status-storage-spec.md [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# Status Storage Specification
```
Comment (oskarth) [inline status-storage-spec.md:1]: I'd also call these e.g. IPFS usage and what not, it isn't a storage spec. Ditto blockchain (Ethereum use?)

File: status-client-spec.md [hunk 13]
```diff
@@ -101,7 +115,7 @@ There are four types of node roles:
 3. Mailservers (servers and clients)
 4. Mobile nodes (Status Clients)
 
-To implement a standard Status client you MUST implement the last node type. The
+To implement a standard Status client you MUST implement both 2. and 4. node types. The
```
Comment (oskarth) [inline status-client-spec.md:105]: Why are we adding 2 ?

File: status-client-spec.md [hunk 14]
```diff
@@ -18,56 +18,70 @@ This document consists of two parts. The first outlines the specifications that
...
+    -   [Appendix A: Security considerations](#appendix-a-security-considerations)
+        -   [Scalability and UX](#scalability-and-ux)
+        -   [Privacy](#privacy)
+        -   [Spam resistance](#spam-resistance)
+        -   [Censorship resistance](#censorship-resistance)
+    -   [Acknowledgements](#acknowledgements)
 
 ## Introduction
 
+The Status mobile app is a multifunction application. This document mainly describes the requirements for establishing connections and communicating between clients. The other features specific to the Status application can be found as follows:
```
Comment (oskarth) [inline status-client-spec.md]: Sorry I don't like this section, it links to no-op work and the story it tells is weird. Document deals with way more than connections, and the main thing that are specified aren't liked. We can reword this into PR once other specs are further along. Removing.

File: status-client-spec.md [hunk 15]
```diff
@@ -56,24 +56,11 @@ have to be implemented in order to be a full Status client. The second gives a d
...
-    -   This spec details how the Status client uses decentralized storage.
--   Decentralized chat
-    -   Decentralized chat is spread across this spec and others that are explained and linked below.
-
 ### Protocol layers
 
-Implementing a Status clients means implementing the following layers. Additionally, there are separate specifications for things like key management and account lifecycle.
+Implementing a Status clients largely means implementing the following layers. Additionally, there are separate specifications for things like key management and account lifecycle.
+
+Some things, such as storage and blockchain usage, are currently underspecified.
```
Comment (oskarth) [inline status-client-spec.md]:
```suggestion
Other aspects, such as how IPFS is used for stickers, how we interact with the Ethereum blockchain or how the browser works, are currently underspecified. These sets of specifications should allow you to implement a a Status client for basic private communication.
```

Comment (oskarth) [reply status-client-spec.md]: cc @corpetty fyi

**Pull request merged**